### ISSUE TITLE
[BUG] Celery becomes unresponsive after Redis reconnect

### ISSUE THREAD

**Issue by DASPRiD** [paperless-ngx/paperless-ngx#6260]
### Description

We performed an upgrade of our Redis instance yesterday at 16:49.

According to the Django logs, the last two task results were performed at 16:40 and 16:51. Looking at the log files, Celery reported at 16:49 that it lost connection to Redis, which matches the update.

At 16:51 Celery reports in the server logs that it successfully connected to Redis again and it processed a task, which matches with the Django logs. After that, no further task processing happened at all (nothing in the Djanog Celery logs) which should happen every ten minutes).

The only other log entry was today at 02:00, this time not about celery but generalls channels_redis.pubsub, where a broken pipe was reported. This is as far as I can see unrelated to the Celery worker issue.

To me it seems that after Celery reconnected to Redis, is performed the pending tasks but did not start looking for new jobs again. When uploading a document today, the upload was just pending on "Upload complete, waiting…". Restarting the Paperless Pod resolved the issue for now.

### Steps to reproduce

1. Stop Redis and restart a while later
2. Observe the log output

### Webserver logs

```bash
[2024-04-01 16:49:30,626] [WARNING] [celery.worker.consumer.consumer] consumer: Connection to broker lost. Trying to re-establish the connection...
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/celery/worker/consumer/consumer.py", line 340, in start
    blueprint.start(self)
  File "/usr/local/lib/python3.11/site-packages/celery/bootsteps.py", line 116, in start
    step.start(parent)
  File "/usr/local/lib/python3.11/site-packages/celery/worker/consumer/consumer.py", line 742, in start
    c.loop(*c.loop_args())
  File "/usr/local/lib/python3.11/site-packages/celery/worker/loops.py", line 97, in asynloop
    next(loop)
  File "/usr/local/lib/python3.11/site-packages/kombu/asynchronous/hub.py", line 373, in create_loop
    cb(*cbargs)
  File "/usr/local/lib/python3.11/site-packages/kombu/transport/redis.py", line 1344, in on_readable
    self.cycle.on_readable(fileno)
  File "/usr/local/lib/python3.11/site-packages/kombu/transport/redis.py", line 569, in on_readable
    chan.handlers[type]()
  File "/usr/local/lib/python3.11/site-packages/kombu/transport/redis.py", line 913, in _receive
    ret.append(self._receive_one(c))
               ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/kombu/transport/redis.py", line 923, in _receive_one
    response = c.parse_response()
               ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/client.py", line 835, in parse_response
    response = self._execute(conn, try_read)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/client.py", line 811, in _execute
    return conn.retry.call_with_retry(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/retry.py", line 49, in call_with_retry
    fail(error)
  File "/usr/local/lib/python3.11/site-packages/redis/client.py", line 813, in <lambda>
    lambda error: self._disconnect_raise_connect(conn, error),
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/client.py", line 800, in _disconnect_raise_connect
    raise error
  File "/usr/local/lib/python3.11/site-packages/redis/retry.py", line 46, in call_with_retry
    return do()
           ^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/client.py", line 812, in <lambda>
    lambda: command(*args, **kwargs),
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/client.py", line 833, in try_read
    return conn.read_response(disconnect_on_error=False, push_request=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/connection.py", line 512, in read_response
    response = self._parser.read_response(disable_decoding=disable_decoding)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/_parsers/hiredis.py", line 124, in read_response
    self.read_from_socket()
  File "/usr/local/lib/python3.11/site-packages/redis/_parsers/hiredis.py", line 86, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
[2024-04-01 16:49:30,628] [WARNING] [py.warnings] /usr/local/lib/python3.11/site-packages/celery/worker/consumer/consumer.py:391: CPendingDeprecationWarning:
In Celery 5.1 we introduced an optional breaking change which
on connection loss cancels all currently executed tasks with late acknowledgement enabled.
These tasks cannot be acknowledged as the connection is gone, and the tasks are automatically redelivered
back to the queue. You can enable this behavior using the worker_cancel_long_running_tasks_on_connection_loss
setting. In Celery 5.1 it is set to False by default. The setting will be set to True by default in Celery 6.0.
  warnings.warn(CANCEL_TASKS_BY_DEFAULT, CPendingDeprecationWarning)
[2024-04-01 16:49:30,631] [ERROR] [celery.worker.consumer.consumer] consumer: Cannot connect to redis://:**@paperless-production-redis-master:6379//: Error 111 connecting to paperless-production-redis-master:6379. Connection refused..
Trying again in 2.00 seconds... (1/100)

[2024-04-01 16:51:42,026] [ERROR] [celery.worker.consumer.consumer] consumer: Cannot connect to redis://:**@paperless-production-redis-master:6379//: Timeout connecting to server.
Trying again in 4.00 seconds... (2/100)

[2024-04-01 16:51:46,038] [INFO] [celery.worker.consumer.connection] Connected to redis://:**@paperless-production-redis-master:6379//

[2024-04-01 16:51:46,047] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[0e1e6314-5fc1-4044-84f2-e257cf186879] received

[2024-04-01 16:51:46,084] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[0e1e6314-5fc1-4044-84f2-e257cf186879] succeeded in 0.0350920669734478s: 'No new documents were added.'

[2024-04-02 02:00:52,903] [ERROR] [channels_redis.pubsub] Unexpected exception in receive task
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/channels_redis/pubsub.py", line 303, in _do_receiving
    message = await self._pubsub.get_message(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/asyncio/client.py", line 1040, in get_message
    response = await self.parse_response(block=(timeout is None), timeout=timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/asyncio/client.py", line 913, in parse_response
    response = await self._execute(
               ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/asyncio/client.py", line 893, in _execute
    return await conn.retry.call_with_retry(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/asyncio/retry.py", line 62, in call_with_retry
    await fail(error)
  File "/usr/local/lib/python3.11/site-packages/redis/asyncio/client.py", line 882, in _disconnect_raise_connect
    raise error
  File "/usr/local/lib/python3.11/site-packages/redis/asyncio/retry.py", line 59, in call_with_retry
    return await do()
           ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/asyncio/connection.py", line 555, in read_response
    raise ConnectionError(f"Error while reading from {host_error} : {e.args}")
redis.exceptions.ConnectionError: Error while reading from paperless-production-redis-master:6379 : (32, 'Broken pipe')
```

### Paperless-ngx version

2.6.3

### Host OS

Ubuntu 22.04 / x86_64

### Installation method

Docker - official image

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

**Comment by stumpylog** [comment]
- we're not the developers of celery.  Actual bugs can be reported to them
- you should restart all the containers, not just Redis.  It's not a surprise that a service going away suddenly causes problems

**Comment by DASPRiD** [comment]
Thanks, I just checked Celery GitHub, and apparently it's a known bug (https://github.com/celery/celery/issues/8384) which was fixed in Celery 5.4.0rc1.
